### PR TITLE
fix: ensure pages _routes.json emulation in dev command handles .s and *s

### LINKED ISSUE #3184: 🐛 BUG: _routes.json emulation incorrect in `wrangler pages dev`
### Which Cloudflare product(s) does this pertain to?

Pages

### What version of `Wrangler` are you using?

2.20.0

### What operating system are you using?

Mac

### Describe the Bug

In wrangler pages dev we emulate how paths in _route.json are matched when deployed to the supercloud.
This emulation is not accurate in the case where you use `.` in a path. For example `"exclude": ["/*.*"]` should only exclude files that have an extension, e.g. "/index.js", but it actually excludes all paths, e.g. "/foo".

Probably needs a fix to `transformRoutingRuleToRegExp()`.

### PR DESCRIPTION
**Pull request by petebacondarwin**
Fixes #3184

**What this PR solves / how to test:**

```
npm run build
cd fixtures/pages-functions-with-routes-app 
npm run test
```

**Author has included the following, where applicable:**

- [x] Tests
- [x] Changeset ([Changeset guidelines](https://github.com/cloudflare/workers-sdk/blob/main/CONTRIBUTING.md#changesets))

**Reviewer is to perform the following, as applicable:**

- Checked for inclusion of relevant tests
- Checked for inclusion of a relevant changeset
- Checked for creation of associated docs updates
- Manually pulled down the changes and spot-tested

### CODE REVIEW

**Comment by petebacondarwin** [comment]
@CarmenPopoviciu and @jahands - I know you worked on this area before. I think what I have changed to here matches how the real-world filter stage works, but I couldn't find a spec or the original code.  Can you take a look?

**Review by RamIdeas — APPROVED** [review]

File: packages/wrangler/templates/pages-dev-util.ts [hunk 1]
```diff
@@ -45,7 +45,10 @@ function transformRoutingRuleToRegExp(rule: string): RegExp {
 	}
 
 	// /foo* => /foo.* => ^/foo.*$
-	transformedRule = `^${transformedRule.replace("*", ".*")}$`;
+	// /*.* => /*\.* => /.*\..* => ^/.*\..*$
+	transformedRule = `^${transformedRule
+		.replaceAll(/\./g, "\\.")
+		.replaceAll(/\*/g, ".*")}$`;
```
Comment (penalosa) [inline packages/wrangler/templates/pages-dev-util.ts:51]: This seems like there might be other gremlins hiding here... What happens if a rule has a `+`? A `()`?

Comment (petebacondarwin) [reply packages/wrangler/templates/pages-dev-util.ts:51]: Yeah I managed to find the internal logic (in C++ 😅) for how this works in production. It is a handcrafted glob implementation.
Given that in that implementation there is no regular expression involved, we should be escaping any regular expression special chars.

Comment (petebacondarwin) [reply packages/wrangler/templates/pages-dev-util.ts:51]: That being said, we have had this logic in place for many months with not many complaints; so I don't think it is a high priority to make this an exact replica of the internal logic right now.

So I'd like to land this to unblock my own work for now, if that's OK.

**Pull request merged**